internal/controller/runtime/fsm/runtime_fsm_create_kubeconfig.go

```diff
@@ -0,0 +1,106 @@
+	}, &cluster)
+
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			m.log.Error(err, "GardenerCluster CR read error", "name", runtimeID)
+			s.instance.UpdateStatePending(imv1.ConditionTypeRuntimeKubeconfigReady, imv1.ConditionReasonKubernetesAPIErr, "False", err.Error())
+			return updateStatusAndStop()
+		}
+
+		m.log.Info("GardenerCluster CR not found, creating a new one", "Name", runtimeID)
+		err = m.Create(ctx, makeGardenerClusterForRuntime(s.instance, s.shoot))
+		if err != nil {
+			m.log.Error(err, "GardenerCluster CR read error", "name", runtimeID)
+			s.instance.UpdateStatePending(imv1.ConditionTypeRuntimeKubeconfigReady, imv1.ConditionReasonKubernetesAPIErr, "False", err.Error())
+			return updateStatusAndStop()
+		}
+
+		m.log.Info("Gardener Cluster CR created, waiting for readiness", "Name", runtimeID)
+		s.instance.UpdateStatePending(imv1.ConditionTypeRuntimeKubeconfigReady, imv1.ConditionReasonGardenerCRCreated, "Unknown", "Gardener Cluster CR created, waiting for readiness")
+		return updateStatusAndRequeueAfter(controlPlaneRequeueDuration)
+	}
+
+	if cluster.Status.State != imv1.ReadyState {
+		m.log.Info("GardenerCluster CR is not ready yet, requeue", "Name", runtimeID, "State", cluster.Status.State)
+		return requeueAfter(controlPlaneRequeueDuration)
+	}
+
+	m.log.Info("GardenerCluster CR is ready", "Name", runtimeID)
+
+	return ensureStatusConditionIsSetAndContinue(&s.instance,
+		imv1.ConditionTypeRuntimeKubeconfigReady,
+		imv1.ConditionReasonGardenerCRReady,
+		"Gardener Cluster CR is ready.",
+		sFnProcessShoot)
+}
+
+func makeGardenerClusterForRuntime(runtime imv1.Runtime, shoot *gardener.Shoot) *imv1.GardenerCluster {
+	gardenCluster := &imv1.GardenerCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GardenerCluster",
+			APIVersion: imv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      runtime.Labels[imv1.LabelKymaRuntimeID],
+			Namespace: runtime.Namespace,
+			Annotations: map[string]string{
+				"skr-domain": *shoot.Spec.DNS.Domain,
+			},
+			Labels: map[string]string{
+				imv1.LabelKymaInstanceID:         runtime.Labels[imv1.LabelKymaInstanceID],
+				imv1.LabelKymaRuntimeID:          runtime.Labels[imv1.LabelKymaRuntimeID],
+				imv1.LabelKymaBrokerPlanID:       runtime.Labels[imv1.LabelKymaBrokerPlanID],
+				imv1.LabelKymaBrokerPlanName:     runtime.Labels[imv1.LabelKymaBrokerPlanName],
+				imv1.LabelKymaGlobalAccountID:    runtime.Labels[imv1.LabelKymaGlobalAccountID],
+				imv1.LabelKymaGlobalSubaccountID: runtime.Labels[imv1.LabelKymaGlobalSubaccountID], // BTW most likely this value will be missing
+				imv1.LabelKymaName:               runtime.Labels[imv1.LabelKymaName],
+
+				// values from Runtime CR fields
+				imv1.LabelKymaPlatformRegion: runtime.Spec.Shoot.PlatformRegion,
+				imv1.LabelKymaRegion:         runtime.Spec.Shoot.Region,
+				imv1.LabelKymaShootName:      shoot.Name,
+
+				// hardcoded values
+				imv1.LabelKymaManagedBy: "lifecycle-manager",
```

Comment:
```suggestion
				imv1.LabelKymaManagedBy: "infrastructure-manager",
```
KEB sets that currently to `lifecycle-manager` which is wrong. KIM is managing this resource. It should be set to `infrastructure-manager`

Reply:
Fixed